src/Services.jl

```diff
@@ -1971,7 +1971,7 @@ function s3(aws::AWSConfig, verb, resource, args=[])
 
     AWSCore.service_rest_xml(
         aws;
-        service      = "s3",
+        service      = get(aws, :service_name, "s3"),
```

Comment:
Looking through this again, the `src/Services.jl` file is generated by `AWSAPI.jl`. To make these changes be proper, we need to modify and special case this in the `AWSAPI.jl` file for the `s3` service.

Reply:
Spent some time with this and trying to get it to work, it seems to be completely busted however.